### ISSUE TITLE
datePicker with mask=true -> ajax not fired when manually change value

### ISSUE THREAD

**Issue by SimonePadovan** [primefaces/primefaces#12754]
### Describe the bug

If i use a datePicker with attribute **mask=true,** the ajax change event is not triggered when i change the date value manually (not selecting it through the calendar).
It works ok if i use PF 14.0.4, but with PF 14.0.5  and PF 14.0.6 it doesnt work

`	<p:datePicker id="event" value="#{localeBean.date8}" pattern="dd/MM/yyyy"  mask="true">
		<p:ajax event="change"	listener="#{localeBean.onDateSelect}" update="msgs" />
	</p:datePicker>`

### PrimeFaces edition

Community

### PrimeFaces version

14.0.6

### JSF implementation

Mojarra

### JSF version

4.0.5

### Java version

17

**Comment by melloware** [comment]
Most likely this fix in InputMask : https://github.com/primefaces/primefaces/issues/12428

**Comment by melloware** [comment]
You can try this Monkeypatch but PF won't be putting this behavior back as it causes too many issues with INputMask.

```js
if (PrimeFaces.widget.DatePicker) {
  PrimeFaces.widget.DatePicker.prototype.applyMask = function() {
    if (this.cfg.inline || this.input.is('[readonly]') || this.input.is(':disabled')) {
      return;
    }
    var $this = this;
    if (this.cfg.mask) {
      var isAutoClear = (this.cfg.maskAutoClear === undefined) ? true : this.cfg.maskAutoClear;
      var maskCfg = {
        placeholder: this.cfg.maskSlotChar || '_',
        clearMaskOnLostFocus: isAutoClear,
        clearIncomplete: isAutoClear,
        autoUnmask: false,
        showMaskOnHover: false,
        onBeforePaste: function(pastedValue, opts) {
          // GitHub #8319 issue with pasting mask
          // TODO: Remove if InputMask 5.0.8+ fixes the issue
          setTimeout(function() {
            $this.input.trigger("input")
          }, 20);
          return pastedValue;
        }
      };
      var pattern = /[mdyhs]/i;
      var isAlias = pattern.test(this.cfg.mask);
      if (isAlias) {
        maskCfg.alias = 'datetime';
        maskCfg.inputFormat = this.cfg.mask;
      } else {
        maskCfg.mask = this.cfg.mask;
      }
      this.input.inputmask('remove').inputmask(maskCfg);
    }
  }
}
```

**Comment by melloware** [comment]
Specifically what I removed was this line: ` this.input.off("blur"); // GitHub #9259/#12428`

**Comment by SimonePadovan** [comment]
Thanks for the answer. I dont understand, are you going to remove mask attribute from datePicker since you're not going to put this behaviour back ?
Does p:inputMask still work as before or even there zou removed onblur event ?
thanks

**Comment by melloware** [comment]
No i mean i am leaving this line in the code `this.input.off("blur"); // GitHub #9259/#12428` which as you can see fixes more issues than it causes.  So everything will work for you if you use the monkeyPatch above which takes out that line but use at your own risk of weird recursion errors.   It may work fine for you i think a lot of the recursion errors depended on the MASK used in InputMask.

**Comment by melloware** [comment]
Please report back how your testing goes.

**Comment by SimonePadovan** [comment]
the monkeyPath works, but i'm not so happy to put such legacy code which can bring to other unexpected behavious using next versions of PF...

**Comment by melloware** [comment]
Yep rolling the dice. But too many people have reported things like `#9259/#12428` to leave in...

**Comment by SimonePadovan** [comment]
Do i have to put a monkeypatch for p:inputMask tag too ? thanks

**Comment by melloware** [comment]
NO yo shouldn't it was specifically a DatePicker with Mask issue as far as I know.

**Comment by melloware** [comment]
PR submitted...Finally got to the bottom of this. We need to Apply Mask and remove its "blur" before we call `Primefaces.skinInput` which adds the proper blur.

**Comment by melloware** [comment]
14.0.10 is out so you should be able to remove your MonkeyPatch

**Comment by SimonePadovan** [comment]
I've tested this issue with PF 14.10.0 and no monkey patch and unfortunatelly it doesn't work.
Ajax listener is not beeing called if i input the data manually.

**Comment by SimonePadovan** [comment]
Here you are
[Uploading mavenproject1.zip…]()

**Comment by melloware** [comment]
it looks like your upload didn't work the link is not right?

**Comment by SimonePadovan** [comment]
[mavenproject1.zip](https://github.com/user-attachments/files/18350095/mavenproject1.zip)

**Comment by melloware** [comment]
thanks!

**Comment by melloware** [comment]
OK here is a PF Test reproducer:
[pf-12754.zip](https://github.com/user-attachments/files/18351013/pf-12754.zip)

Run `mvn clean jetty:run`

**Comment by SimonePadovan** [comment]
Thank u very much Melloware

Il mer 8 gen 2025, 22:17 Melloware ***@***.***> ha scritto:

**Comment by melloware** [comment]
OK 14.0.11 is released

**Comment by SimonePadovan** [comment]
Thank you very much!

Il mar 28 gen 2025, 20:26 Melloware ***@***.***> ha scritto:

**Comment by SimonePadovan** [comment]
@melloware I'm sorry but even with this version (14.0.11) the problem persists and has not been resolved.
When i type the date manually (not selecting it through the button) the listener is not beeing called.

**Comment by melloware** [comment]
@SimonePadovan can you double check that you have removed all previous MonkeyPatches?  Those old patches will actually break the new code

**Comment by SimonePadovan** [comment]
Hi, i've just doubled checked it. I confirm that i've removed the monkeypatch
